zeek/Reading-Zeek-Log-Formats.md

```diff
@@ -0,0 +1,158 @@
+    service: null (bstring),
+    duration: 4.266ms,
+    orig_bytes: 97 (uint64),
+    resp_bytes: 19 (uint64),
+    conn_state: "RSTR" (bstring),
+    local_orig: null (bool),
+    local_resp: null (bool),
+    missed_bytes: 0 (uint64),
+    history: "ShADTdtr" (bstring),
+    orig_pkts: 10 (uint64),
+    orig_ip_bytes: 730 (uint64),
+    resp_pkts: 6 (uint64),
+    resp_ip_bytes: 342 (uint64),
+    tunnel_parents: null (1=(|[bstring]|))
+} (=2)
+```
+
+Other than Zed, Zeek provides one of the richest data typing systems available
+and therefore such records typically need no adjustment to their data types
+once they've been read in as is. The
+[Zed/Zeek Data Type Compatibility](Data-Type-Compatibility.md) document
+provides further detail on how the rich data types in Zeek TSV map to the
+equivalent [rich types in Zed](../docs/formats/zson.md#33-primitive-values).
+
+# Zeek NDJSON
+
+As an alternative to the default TSV format, there are two common ways that
+Zeek may instead generate logs in [NDJSON](http://ndjson.org/) format.
+
+1. Using the [JSON Streaming Logs](https://github.com/corelight/json-streaming-logs)
+   package (recommended for use with Zed)
+2. Using the built-in [ASCII logger](https://docs.zeek.org/en/current/scripts/base/frameworks/logging/writers/ascii.zeek.html)
+   configured with `redef LogAscii::use_json = T;`
+
+In both cases, Zed tools such as `zq` can read these NDJSON logs automatically
+as is, but with caveats.
+
+Let's revisit the same `conn` record we just examined from the Zeek TSV log,
+but now using the
+[Zeek NDJSON zed-sample-data](https://github.com/brimdata/zed-sample-data/tree/main/zeek-ndjson),
+which was generated using the JSON Streaming Logs package.
+
+#### Example:
+
+```zq-command zed-sample-data/zeek-ndjson
+zq -Z 'head 1' conn.ndjson.gz
+```
+
+#### Output:
+```zq-output
+{
+    _path: "conn",
+    _write_ts: "2018-03-24T17:15:21.400275Z",
+    ts: "2018-03-24T17:15:21.255387Z",
+    uid: "C8Tful1TvM3Zf5x8fl",
+    "id.orig_h": "10.164.94.120",
+    "id.orig_p": 39681,
+    "id.resp_h": "10.47.3.155",
+    "id.resp_p": 3389,
+    proto: "tcp",
+    duration: 0.004266023635864258,
+    orig_bytes: 97,
+    resp_bytes: 19,
+    conn_state: "RSTR",
+    missed_bytes: 0,
+    history: "ShADTdtr",
+    orig_pkts: 10,
+    orig_ip_bytes: 730,
+    resp_pkts: 6,
+    resp_ip_bytes: 342
+}
+```
+
+When we compare this to the TSV example, we notice a few things right away that
+all follow from the records having been previously output as JSON.
+
+1. The timestamps like `_write_ts` and `ts` are printed as strings rather than
+   the ZSON `time` type
+2. The IP addresses such as `id.orig_h` and `id.resp_h` are printed as strings
+   rather than the ZSON `ip` type
+3. The connection `duration` is printed as a floating point number rather than
+   the ZSON `duration` type
+4. The keys for the null-valued fields in the record read from
+   TSV are not present in the record read from NDJSON
```

Comment:
All the guidance I've found suggests that if any item in a vertical list is a full sentence, then all items should be punctuated as full sentences.
```suggestion
1. The timestamps like `_write_ts` and `ts` are printed as strings rather than
   the ZSON `time` type.
2. The IP addresses such as `id.orig_h` and `id.resp_h` are printed as strings
   rather than the ZSON `ip` type.
3. The connection `duration` is printed as a floating point number rather than
   the ZSON `duration` type.
4. The keys for the null-valued fields in the record read from
   TSV are not present in the record read from NDJSON.
```